src/spells.cpp

```diff
@@ -236,7 +236,8 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 
 InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
-	auto& it = std::advance(instants.begin(), spellId);
+	auto& it = instants.begin();
+	std::advance(it, spellId);
```

Comment:
I'm not sure but if you use advance past the container's end. It will most likely crash when you do it->second. Maybe change it to a for loop (std::advance and std::next do that)